User info class

SakiAkizuki opened the pull request.

Commit:
Confirmation Page Update

Commit:
uw account only

Commit:
Final version

Commit:
setupProfile update the user info

Commit:
real time update the drawer bar

SakiAkizuki committed:
Merge branch 'master' into UserInfoClass

Leksha commented on StudyBuddy/app/src/main/java/uw/studybuddy/Events/EventCreation.java:
```diff
@@ -89,6 +89,31 @@ public void onClick(View v) {
+            dataMap.put("subject", subject);
+
+
+            mDatabase.push().setValue(dataMap).addOnCompleteListener(new OnCompleteListener<Void>() {
```
We wanted to keep all database-related methods in the `FirebaseInstance` class. We already have a method in `FirebaseInstance` called `addNewEventToDatabase`that does this.

Leksha commented on StudyBuddy/app/src/main/java/uw/studybuddy/MainActivity.java:
```diff
@@ -41,7 +56,13 @@
-    private UserInfo user;
+    private UserInfo user = new UserInfo();
+
+    DatabaseReference mUserRootRef = FirebaseDatabase.getInstance().getReference().child("Users")
```
Refactor to `FirebaseInstance` class.

SakiAkizuki replied:
the mUserRootRef have to set here for the value change listener function

Leksha replied:
I meant that this code `FirebaseDatabase.getInstance().getReference().child("Users")` should be in `FirebaseInstance` class

We don't want any database-related code like this part `.child("Users")`in other places.

Leksha commented on StudyBuddy/app/src/main/res/layout/activity_register.xml:
```diff
@@ -40,7 +40,7 @@
         android:ems="10"
         android:fontFamily="sans-serif-smallcaps"
-        android:hint="Confirmation email address  "
+        android:hint="Please Type in Your UW Ques ID"
```
- Quest 
- Just use the `EmptyEmail` string instead

SakiAkizuki committed:
delete the command part

/*
        if(!TextUtils.isEmpty(course) && !TextUtils.isEmpty(description) && !TextUtils.isEmpty(location) && !TextUtils.isEmpty(subject)){
            dataMap.put("course", course);
            dataMap.put("description", description);
            dataMap.put("location", location);
            dataMap.put("subject", subject);


            mDatabase.push().setValue(dataMap).addOnCompleteListener(new OnCompleteListener<Void>() {
                @Override
                public void onComplete(@NonNull Task<Void> task) {
                    if(task.isSuccessful()){
                        Toast.makeText(EventCreation.this, "Saving information...",Toast.LENGTH_LONG).show();
                    } else {
                        Toast.makeText(EventCreation.this, "Error occured.",Toast.LENGTH_LONG).show();
                    }
                }
            });

        } else {
            Toast.makeText(EventCreation.this, "You have unfilled blank.",Toast.LENGTH_LONG).show();
        }*/

SakiAkizuki committed:
Quest

Quest
